### PR TITLE
[core] Constant strides can be get from bounds tensor

### PR DESCRIPTION
**Pull request by praasz**
### Details:
 - Constant has atwo additional members functions:
   - `get_strides` to get Constant data strides (throw exception for low precision data)
   - `get_tensor_view` to get Constant data as ov::Tensor view.

### Tickets:
 - [CVS-143575](https://jira.devtools.intel.com/browse/CVS-143575)

### CODE REVIEW

**Review by ilya-lavrenov — APPROVED** [review]

**Review by jane-intel — COMMENTED** [review]
Can we provide `Constant::get_tensor` or `Constant::get_stride` to solve the issue?
Why did we choose to incorporate it in partial value propagation?

**Review by slyalin — CHANGES_REQUESTED** [review]
I find it strange to workaround missing methods in Constant class with re-use of the value bound functionality. Why not to add the required methods in Constant itself? You still need to verify that you are working with Constant node first before getting strides (otherwise they are useless, because bound tensor values are not connected to a specific constant). So the natural approach is to extend Constant node.

**Comment by praasz** [comment]
> I find it strange to workaround missing methods in Constant class with re-use of the value bound functionality. Why not to add the required methods in Constant itself? You still need to verify that you are working with Constant node first before getting strides (otherwise they are useless, because bound tensor values are not connected to a specific constant). So the natural approach is to extend Constant node.

I've added dedicated members to Constant.

@jane-intel 
Changes related partial value evaluation will move to other PR to not mix it with get strides, it can be useful as will remove not necessary data copy and evaluation should be faster.

**Review by jane-intel — APPROVED** [review]
Thank you, this looks good!

File: src/core/src/op/constant.cpp [hunk 1]
```diff
@@ -483,6 +503,11 @@ void Constant::update_identical_flags(bool is_checked, bool identical_value) con
 
 void Constant::validate_and_infer_types() {
     set_output_type(0, m_element_type, m_shape);
+    if (const auto tensor = get_tensor_view()) {
+        auto& output_tensor = get_output_tensor(0);
+        output_tensor.set_lower_value(tensor);
+        output_tensor.set_upper_value(tensor);
+    }
```
Comment (jane-intel) [inline src/core/src/op/constant.cpp:510]:
```suggestion
```

File: src/core/tests/constant.cpp [hunk 2]
```diff
@@ -1815,6 +1835,8 @@ TEST(constant, float8_e4m3_vector) {
 
     ov::op::v0::Constant const_op_from_ptr(ov::element::f8e4m3, data_shape, data_vec.data());
     EXPECT_EQ(data_vec, const_op_from_ptr.get_vector<ov::float8_e4m3>());
+    EXPECT_EQ(const_op_from_ptr.get_output_tensor(0).get_upper_value().get_strides(),
+              Strides({element::f8e4m3.size()}));
```
Comment (jane-intel) [inline src/core/tests/constant.cpp:1839]:
```suggestion
```

File: src/core/tests/constant.cpp [hunk 3]
```diff
@@ -1838,6 +1860,8 @@ TEST(constant, float8_e5m3_vector) {
 
     ov::op::v0::Constant const_op_from_ptr(ov::element::f8e5m2, data_shape, data_vec.data());
     EXPECT_EQ(data_vec, const_op_from_ptr.get_vector<ov::float8_e5m2>());
+    EXPECT_EQ(const_op_from_ptr.get_output_tensor(0).get_upper_value().get_strides(),
+              Strides({element::f8e5m2.size()}));
```
Comment (jane-intel) [inline src/core/tests/constant.cpp:1864]:
```suggestion
```

**Pull request merged**